bumble/hid.py

```diff
@@ -0,0 +1,246 @@
+
+    async def interrupt_channel_disconnect(self):
+        await self.l2cap_intr_channel.disconnect()  # type: ignore
+
+    async def control_channel_disconnect(self):
+        await self.l2cap_ctrl_channel.disconnect()  # type: ignore
+
+    def on_connection(self, l2cap_channel: l2cap.Channel) -> None:
+        logger.debug(f'+++ New L2CAP connection: {l2cap_channel}')
+        l2cap_channel.on('open', lambda: self.on_l2cap_channel_open(l2cap_channel))
+
+    def on_l2cap_channel_open(self, l2cap_channel: l2cap.Channel) -> None:
+        if l2cap_channel.psm == HID_CTRL_PSM:
+            self.l2cap_ctrl_channel = l2cap_channel
+            self.l2cap_ctrl_channel.sink = self.on_ctrl_pdu
+        else:
+            self.l2cap_intr_channel = l2cap_channel
+            self.l2cap_intr_channel.sink = self.on_intr_pdu
+        logger.debug(f'$$$ L2CAP channel open: {l2cap_channel}')
+
+    def on_ctrl_pdu(self, pdu: bytes) -> None:
+        logger.debug(f'<<< HID CONTROL PDU: {pdu.hex()}')
+        # Here we will receive all kinds of packets, parse and then call respective callbacks
+        message_type = pdu[0] >> 4
+        param = pdu[0] & 0x0f
+        if (message_type == HID_HANDSHAKE):
+            logger.debug('<<< HID HANDSHAKE')
+            self.handle_handshake(param)
+            self.emit('handshake', )
+        elif (message_type == HID_DATA):
+            logger.debug('<<< HID CONTROL DATA')
+            self.emit('data', pdu)
+        elif (message_type == HID_CONTROL):
+            if (param == HID_SUSPEND):
+                logger.debug('<<< HID SUSPEND')
+                self.emit('suspend', pdu)
+            elif (param == HID_EXIT_SUSPEND):
+                logger.debug('<<< HID EXIT SUSPEND')
+                self.emit('exit_suspend', pdu)
+            elif (param == HID_VIRTUAL_CABLE_UNPLUG):
+                logger.debug('<<< HID VIRTUAL CABLE UNPLUG')
+                self.emit('virtual_cable_unplug')
+            else:
+                logger.debug('<<< HID CONTROL OPERATION UNSUPPORTED')
+        else:
+            logger.debug('<<< HID CONTROL DATA')
+            self.emit('data', pdu)
+
+    def on_intr_pdu(self, pdu: bytes) -> None:
+        logger.debug(f'<<< HID INTERRUPT PDU: {pdu.hex()}')
+        self.emit("data", pdu)
+
+    def register_data_cb(self, data_cb):
+        self.on('data', data_cb)
+
+    def register_handshake_cb(self, handshake_cb):
+        self.on('handshake', handshake_cb)
+
+    def register_virtual_cable_unplug(self, virtual_cable_unplug_cb):
+        self.on('virtual_cable_unplug', virtual_cable_unplug_cb)
+
+    def get_report(self, report_type, report_id, buffer_size):
+        if(report_type == HID_OTHER_REPORT):
+            param = report_type
+        else:
+            param = 0x08 | report_type
+        header = ((HID_GET_REPORT << 4) | param)
+        msg = bytes([header, report_id, (buffer_size & 0xff), ((buffer_size >> 8) & 0xff)])
```

Comment:
Looks like a packet format. Can it be a packet class?

Reply:
done HIDPacket class

Reply:
The BT spec calls those Message. Please use `class Message` rather than `class HIDPacket`